subsys/net/ip/tcp.c

```diff
@@ -352,6 +348,32 @@ static void tcp_send(struct net_pkt *pkt)
 	tcp_pkt_unref(pkt);
 }
 
+static int tcp_derive_rto(struct tcp *conn)
+{
+	int rto;
+
+#ifdef CONFIG_NET_TCP_RANDOMIZED_RTO
+	uint32_t gain;
+
+	gain = (uint32_t)conn->rto_random;
+	gain += 1 << 9;
+
+	rto = (gain * CONFIG_NET_TCP_INIT_RETRANSMISSION_TIMEOUT) >> 9;
+#else
+	rto = CONFIG_NET_TCP_INIT_RETRANSMISSION_TIMEOUT;
+#endif
+
+	/* The last retransmit does not need to wait that long */
+	if (conn->send_data_retries < tcp_retries) {
+		/* Every retransmit, the retransmission timeout increases by a factor 1.5 */
```

Comment:
If I understand correctly, the initial random retransmission timeout will be in range of `(CONFIG_NET_TCP_INIT_RETRANSMISSION_TIMEOUT; CONFIG_NET_TCP_INIT_RETRANSMISSION_TIMEOUT * 1.5)`.

I don't understand though, why is the consecutive retransmission delay multiplied by 1.5 instead of simply being doubled? It complicates the logic a bit (i. e. need to use this loop instead of a single bitshift).

Reply:
The 1.5 is a bit of a trade-off, the "lost" time is less, secondly it makes passing the tests much easier, otherwise the timeout very quickly becomes large.
It complicates the computation quite a bit indead.

Reply:
Ok, I guess we can live with that, the computation time would only affect further retransmissions anyway and I guess we don't have to worry about performance that much at this point.